src/StatusQ/Controls/StatusTagSelector.qml

```diff
@@ -0,0 +1,109 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+//TODO check colors, sizes, margins, general clean up
+Item {
+    id: root
+
+    implicitWidth: 448
+    implicitHeight: 44
+
+    property alias text: edit.text
+    property ListModel namesModel: ListModel { }
+
+    Rectangle {
+        anchors.fill: parent
+        radius: 8
+        color: Theme.palette.baseColor2
+        border.width: 1
+        border.color: edit.activeFocus ? Theme.palette.primaryColor2 : "transparent"
+
+        RowLayout {
+            anchors.fill: parent
+            anchors.leftMargin: 16
+            anchors.rightMargin: 16
+            spacing: 8
+            StatusBaseText {
+                Layout.preferredWidth: 22
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+                color: Theme.palette.baseColor1
+                text: "To: "
+            }
+
+            ListView {
+                id: namesList
+                //TODO check if 220 is the correct limit for width
+                Layout.preferredWidth: (childrenRect.width > 220) ? 220 : childrenRect.width
+                implicitHeight: 30
+                visible: (count > 0)
+                Layout.alignment: Qt.AlignVCenter
+                model: namesModel
+                orientation: ListView.Horizontal
+                spacing: 8
+                clip: true
+                onContentWidthChanged: {
+                    positionViewAtEnd();
+                }
+
+                delegate: Rectangle {
+                    id: nameDelegate
+                    width: (nameText.contentWidth + 40)
+                    height: 30
+                    color: Theme.palette.primaryColor1
+                    radius: 8
+                    StatusBaseText {
+                        id: nameText
+                        anchors.left: parent.left
+                        anchors.leftMargin: 8
+                        anchors.verticalCenter: parent.verticalCenter
+                        color: Theme.palette.white
+                        text: name
+                    }
+                    ToolButton {
+                        width: 24
+                        height: 24
+                        anchors.verticalCenter: parent.verticalCenter
+                        anchors.left: nameText.right
+                        background: null
+                        contentItem: StatusIcon {
+                            color: Theme.palette.white
```

Comment:
Does this work across themes?

Reply:
I see all status components are using this, is this not supposed to be working?

Reply:
Ah, my point was rather that, what is `white` here should probably not be `white` across all themes. Judging by other designs and components (e.g. StatusBadge), this should most likely become black in dark theme. So it has to be `Theme.palette.indirectColor`.

Reply:
done